gumby/sync.py

```diff
@@ -193,7 +209,8 @@ def setConnectionMade(self, proto):
             if self._made_looping_call and self._made_looping_call.running:
                 self._made_looping_call.stop()
 
-            self.pushIdToSubscribers()
+            # this gives the reactor some seconds to process pending id request messages
+            reactor.callLater(5 + 0.2*self.expected_subscribers, self.pushIdToSubscribers)
```

Comment:
What happens if we run an experiment with 1000 nodes? Then I would expect the `self.expected_subscribers` array to contain 1000 elements, so I have to wait at most 205 seconds?

Reply:
Yes. Perhaps we can taper this slightly to increase the base 5 to 10 and reduce the 0.2 to 0.05? (== 60 seconds for 1000 nodes) But I didn't have the scale to test this properly.

Since I implemented requesting an ID as an optional feature, I couldn't make pushId..() depend on messages that might never come. However as I said in the OP, when running this on the DAS, a little more work is needed. Each process_guard will allocate the same block of ID's. So I wasn't really thinking about the 1000 nodes scenario. (If that really exists, we might want to check the default port allocations on dispersy too. That is set up with max a few hundred in mind currently)